experimental/language-generation/docs/api-reference.md

```diff
@@ -88,65 +78,119 @@ public string Content { get; set; }
+    IList<Templates> references = null,
+    string content = null,
+    string id = null,
+    ExpressionParser expressionParser = null,
+    ImportResolverDelegate importResolver = null,
+    IList<string> options = null)
 {
-    Templates = templates ?? new List<LGTemplate>();
-    Imports = imports ?? new List<LGImport>();
+    if (templates != null)
+    {
+        this.AddRange(templates);
+    }
+
+    Imports = imports ?? new List<TemplateImport>();
     Diagnostics = diagnostics ?? new List<Diagnostic>();
-    References = references ?? new List<LGFile>();
+    References = references ?? new List<Templates>();
     Content = content ?? string.Empty;
     ImportResolver = importResolver;
     Id = id ?? string.Empty;
-    ExpressionEngine = expressionEngine ?? new ExpressionEngine();
+    ExpressionParser = expressionParser ?? new ExpressionParser();
+    Options = options ?? new List<string>();
 }
 ```
 
 #### Methods
 
 ```C#
+/// <summary>
+/// Parser to turn lg content into a <see cref="LanguageGeneration.Templates"/>.
+/// </summary>
+/// <param name="filePath"> absolut path of a LG file.</param>
+/// <param name="importResolver">resolver to resolve LG import id to template text.</param>
+/// <param name="expressionParser">expressionEngine Expression engine for evaluating expressions.</param>
+/// <returns>new <see cref="LanguageGeneration.Templates"/> entity.</returns>
+public static Templates ParseFile(
+    string filePath,
+    ImportResolverDelegate importResolver = null,
+    ExpressionParser expressionParser = null);
+
+/// <summary>
+/// Parser to turn lg content into a <see cref="LanguageGeneration.Templates"/>.
+/// </summary>
+/// <param name="content">Text content contains lg templates.</param>
+/// <param name="id">id is the identifier of content. If importResolver is null, id must be a full path string. </param>
+/// <param name="importResolver">resolver to resolve LG import id to template text.</param>
+/// <param name="expressionParser">expressionEngine parser engine for parsing expressions.</param>
+/// <returns>new <see cref="LanguageGeneration.Templates"/> entity.</returns>
+public static Templates ParseText(
+    string content,
+    string id = "",
+    ImportResolverDelegate importResolver = null,
+    ExpressionParser expressionParser = null);
+
 /// <summary>
 /// Evaluate a template with given name and scope.
 /// </summary>
 /// <param name="templateName">Template name to be evaluated.</param>
 /// <param name="scope">The state visible in the evaluation.</param>
 /// <returns>Evaluate result.</returns>
-public object EvaluateTemplate(string templateName, object scope = null)
-```
+public object Evaluate(string templateName, object scope = null);
+
+/// <summary>
+/// Use to evaluate an inline template str.
+/// </summary>
+/// <param name="text">inline string which will be evaluated.</param>
+/// <param name="scope">scope object or JToken.</param>
+/// <returns>Evaluate result.</returns>
```

Comment:
I believe our standard is to Upper case the sentence for all of these which is probably an issue in the code.  This is also pretty minimal on what "evaluating text" means, i.e. ${} are evaluated.